### PR TITLE
Fix discovery at /apis. Splitting #14668.

### PR DESCRIPTION
**Pull request by caesarxuchao**
Split #14668 to smaller pieces. This PR is piece 1. 
It does:
1. fix the error that API server returns "group/group/version" at /apis
2. add a unit test to prevent such error occurs again

This PR doesn't clean the variable names ("version") that mismatch their contents ("group/version"). That will be done after 1.1.

@lavalamp @deads2k

### CODE REVIEW

File: pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -78,13 +78,16 @@ type Mux interface {
 type APIGroupVersion struct {
 	Storage map[string]rest.Storage
 
-	Root    string
+	Root string
+	// TODO: caesarxuchao: Version actually contains "group/version", refactor it to avoid confusion.
 	Version string
 
 	// ServerVersion controls the Kubernetes APIVersion used for common objects in the apiserver
 	// schema like api.Status, api.DeleteOptions, and api.ListOptions. Other implementors may
```
Comment (lavalamp) [inline pkg/apiserver/apiserver.go]: From this description, now that we have unversioned objects, we don't need this variable at all. Just noting this for the future.

**Comment by lavalamp** [comment]
LGTM once you make travis happy & squash.

Thank you so much for an easy to review PR :) Please cherry pick this.

File: pkg/apiserver/apiserver.go (hunk 1 above)
Comment (caesarxuchao) [inline pkg/apiserver/apiserver.go]: Will we move DeleteOptions and ListOptions to unversioned as well?

File: pkg/apiserver/api_installer.go [hunk 2]
```diff
@@ -90,7 +90,8 @@ func (a *APIInstaller) Install(ws *restful.WebService) (apiResources []api.APIRe
 func (a *APIInstaller) NewWebService() *restful.WebService {
 	ws := new(restful.WebService)
 	ws.Path(a.prefix)
-	ws.Doc("API at " + a.prefix + " version " + a.group.Version)
+	// a.prefix contains "prefix/group/version"
```
Comment (caesarxuchao) [inline pkg/apiserver/api_installer.go]: @lavalamp this is a new change after your last review. PTAL. Thank you!

**Comment by caesarxuchao** [comment]
@k8s-bot test this please

**Comment by caesarxuchao** [comment]
@k8s-bot test this

File: pkg/apiserver/apiserver.go (hunk 1 above)
Comment (lavalamp) [inline pkg/apiserver/apiserver.go]: oh, good question. Let's look into that later.

File: pkg/apiserver/api_installer.go (hunk 2 above)
Comment (lavalamp) [inline pkg/apiserver/api_installer.go]: LGTM

**Comment by caesarxuchao** [comment]
e2e flake:

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/container_probe.go:74
pod never became ready
Expected error:
    <*errors.errorString | 0xc20811aa70>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
not to have occurred

**Comment by caesarxuchao** [comment]
@k8s-bot test this please

**Comment by caesarxuchao** [comment]
adding LGTM per @lavalamp's comment

**Comment by lavalamp** [comment]
@caesarxuchao Please make a cherry pick.

**Pull request merged**